### PR TITLE
Tweak wording in value type docs

### PR DESCRIPTION
**Pull request by NunoFilipeSantos**
- User-defined were written inconsistently (with/without hyphen)
- Rewording some sentences for user clarity.

### CODE REVIEW

File: docs/types/value-types.rst [hunk 1]
```diff
@@ -47,7 +46,7 @@ access the minimum and maximum value representable by the type.
 
   Integers in Solidity are restricted to a certain range. For example, with ``uint32``, this is ``0`` up to ``2**32 - 1``.
   There are two modes in which arithmetic is performed on these types: The "wrapping" or "unchecked" mode and the "checked" mode.
-  By default, arithmetic is always "checked", which mean that if the result of an operation falls outside the value range
+  By default, arithmetic is always "checked", meaning that if an operation's result falls outside the value range
```
Comment (chriseth) [inline docs/types/value-types.rst]:
```suggestion
  By default, arithmetic is always "checked", meaning that if an operation's result falls outside the value range
```

**Review by chriseth — COMMENTED** [review]
Good apart from the superfluous space. I think it is better with hyphen.

**Review by cameel — COMMENTED** [review]
Looks good other than the superfluous whitespace pointed out by @chriseth.

Looks like some tests are failing but that's some infrastructure problem on CircleCI side. I've seen it on other PRs today as well.

**Comment by NunoFilipeSantos** [comment]
Thanks! I was reading the page and found the wording confusing, so I tried to fix it.

File: docs/types/value-types.rst [hunk 2]
```diff
@@ -182,7 +181,7 @@ Operators:
 Address
 -------
 
-The address type comes in two flavours, which are largely identical:
+The address type comes in two largely identical flavors:
```
Comment (cameel) [inline docs/types/value-types.rst:184]: Also, we generally use British spelling :) Though I usually forget about it myself - I'm just too used to the US one.
```suggestion
The address type comes in two largely identical flavours:
```

**Review by cameel — APPROVED** [review]

File: docs/types/value-types.rst (hunk 2 above)
Comment (NunoFilipeSantos) [reply docs/types/value-types.rst:184]: We do??? I've read in multiple places words written in international English (US).

Comment (cameel) [reply docs/types/value-types.rst:184]: I mean, we have a rule. Whether we're applying it consistently is another matter.

I often catch myself not remembering about it myself, which is why I approved the PR anyway :)

**Comment by NunoFilipeSantos** [comment]
Color: https://docs.soliditylang.org/en/v0.8.18/brand-guide.html
License: https://docs.soliditylang.org/en/v0.8.18/search.html?q=license&check_keywords=yes&area=default#
Analize: https://docs.soliditylang.org/en/v0.8.18/search.html?q=analyze&check_keywords=yes&area=default

and goes on and on...

File: docs/types/value-types.rst (hunk 2 above)
Comment (NunoFilipeSantos) [reply docs/types/value-types.rst:184]: Well, I kind of challenge that "rule". British is not International English. 🔥

Comment (cameel) [reply docs/types/value-types.rst:184]: I would not be against not using British English :) But the law requires me to point this out as a typo :P

**Pull request merged**